diff --git a/ngx_http_upload_module.c b/ngx_http_upload_module.c
index 14521dc..2cac1d5 100644
--- a/ngx_http_upload_module.c
+++ b/ngx_http_upload_module.c
@@ -777,7 +777,7 @@ ngx_http_upload_handler(ngx_http_request_t *r)
     if(r->method & NGX_HTTP_OPTIONS)
         return ngx_http_upload_options_handler(r);
 
-    if (!(r->method & NGX_HTTP_POST))
+    if (!(r->method & (NGX_HTTP_POST | NGX_HTTP_PUT)))
         return NGX_HTTP_NOT_ALLOWED;
 
     ulcf = ngx_http_get_module_loc_conf(r, ngx_http_upload_module);
