diff --git a/magick/version.h b/magick/version.h
index f189553..e921f5f 100644
--- a/magick/version.h
+++ b/magick/version.h
@@ -28,7 +28,7 @@ extern "C" {
 #define MagickPackageName "ImageMagick"
 #define MagickCopyright  "Copyright (C) 1999-2012 ImageMagick Studio LLC"
 #define MagickSVNRevision  "8971"
-#define MagickLibVersion  0x679
+#define MagickLibVersion  0x677
 #define MagickLibVersionText  "6.7.9"
 #define MagickLibVersionNumber  5,0,0
 #define MagickLibAddendum  "-0"
diff --git a/version.sh b/version.sh
index 6c4d7ba..5630ea7 100644
--- a/version.sh
+++ b/version.sh
@@ -12,7 +12,7 @@ PACKAGE_NAME='ImageMagick'
 # PACKAGE_NAME (e.g. "1.0.0").
 PACKAGE_VERSION='6.7.9'
 PACKAGE_PERL_VERSION='6.79'
-PACKAGE_LIB_VERSION="0x679"
+PACKAGE_LIB_VERSION="0x677"
 PACKAGE_RELEASE="0"
 PACKAGE_LIB_VERSION_NUMBER="6,7,9,${PACKAGE_RELEASE}"
 PACKAGE_RELEASE_DATE=`date +%F`
