--- node-0.6.8/src/node_http_parser.cc	2012-05-07 14:35:09.000000000 -0700
+++ node-0.6.8-http-fix/src/node_http_parser.cc	2012-05-07 14:42:45.000000000 -0700
@@ -191,7 +191,7 @@ struct StringPtr {
   void Update(const char* str, size_t size) {
     if (str_ == NULL)
       str_ = str;
-    else if (on_heap_ || str_ + size != str) {
+    else if (on_heap_ || str_ + size_ != str) {
       // Non-consecutive input, make a copy on the heap.
       // TODO Use slab allocation, O(n) allocs is bad.
       char* s = new char[size_ + size];
